### PR TITLE
fix: update pilot 2024 rollout logic

### PR DESCRIPTION
**Pull request by ReubenRobertsHEE**
Note: placement changes are transitive on PM changes.

TIS21-6564
TIS21-6562

### CODE REVIEW

File: src/main/java/uk/nhs/hee/trainee/details/service/ProgrammeMembershipService.java [hunk 1]
```diff
@@ -310,37 +299,15 @@ public boolean isPilot2024(String traineeTisId, String programmeMembershipId) {
     String managingDeanery = programmeMembership.getManagingDeanery();
     LocalDate startDate = programmeMembership.getStartDate();
     LocalDate dayBefore01082024 = LocalDate.of(2024, 7, 31);
-    LocalDate dayAfter31102024 = LocalDate.of(2024, 11, 1);
     if ((PILOT_2024_LOCAL_OFFICES_ALL_PROGRAMMES.stream()
         .anyMatch(lo -> lo.equalsIgnoreCase(managingDeanery)))
-        && (startDate.isAfter(dayBefore01082024) && startDate.isBefore(dayAfter31102024))) {
+        && (startDate.isAfter(dayBefore01082024))) {
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/trainee/details/service/ProgrammeMembershipService.java:304]: This needs updating to 2024-11-01

**Review by DorisWongNHS — DISMISSED** [review]

File: src/main/java/uk/nhs/hee/trainee/details/service/ProgrammeMembershipService.java [hunk 2]
```diff
@@ -310,37 +299,15 @@ public boolean isPilot2024(String traineeTisId, String programmeMembershipId) {
...
-        && (startDate.isAfter(dayBefore01082024) && startDate.isBefore(dayAfter31082024))
-        && (programmeMembership.getCurricula().stream().anyMatch(c ->
-        PILOT_2024_NW_SPECIALTIES.stream().anyMatch(
-            s -> s.equalsIgnoreCase(c.getCurriculumSpecialty())))
-        || programmeMembership.getProgrammeName()
-        .equalsIgnoreCase("Cardio-thoracic surgery (run through)")
-        || programmeMembership.getProgrammeName()
-        .equalsIgnoreCase("Oral and maxillo-facial surgery (run through)"));
+    LocalDate dayBefore01022025 = LocalDate.of(2025, 1, 31);
+    return (managingDeanery.equalsIgnoreCase("Thames Valley")
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/trainee/details/service/ProgrammeMembershipService.java:309]: We could simplify and merge with the above condition to eliminiate the `PILOT_2024_LOCAL_OFFICES_ALL_PROGRAMMES` list.

```java
LocalDate notificationsEpoch = managingDeanery.equalsIgnoreCase("Thames Valley") ? LocalDate.of(2025, 1, 31) : LocalDate.of(2024, 10, 31)

return startDate.isAfter(notificationsEpoch);
```

**Review by Judge40 — CHANGES_REQUESTED** [review]

File: src/main/java/uk/nhs/hee/trainee/details/service/ProgrammeMembershipService.java (hunk 1 above)
Comment (ReubenRobertsHEE) [reply src/main/java/uk/nhs/hee/trainee/details/service/ProgrammeMembershipService.java:304]: I might be a bit confused, but I put some notes on this ticket: https://hee-tis.atlassian.net/browse/TIS21-6565

File: src/main/java/uk/nhs/hee/trainee/details/service/ProgrammeMembershipService.java (hunk 2 above)
Comment (ReubenRobertsHEE) [reply src/main/java/uk/nhs/hee/trainee/details/service/ProgrammeMembershipService.java:309]: We still need to exclude North East though, which is why I kept the list

**Pull request closed without merging**